### PR TITLE
Improve dashboard as code

### PR DESCRIPTION
**Pull request by JCZuurmond**
- Fix failing integration test
- Increase unit test coverage
- Minimize integration test surface
- Reduce code duplication
- Use dataclasses methods
- Refactor format query

### CODE REVIEW

File: src/databricks/labs/lsql/dashboards.py [hunk 1]
```diff
@@ -24,41 +25,59 @@
...
-            page = first_page.as_dict()
-            yaml.safe_dump(page, f)
-        assert True
-
-    def create_dashboard(self, dashboard_folder: Path) -> Dashboard:
+            query = self.format_query(dataset.query)
+            with (local_path / f"{dataset.name}.sql").open("w") as f:
+                f.write(query)
+
+        for page in dashboard.pages:
```
Comment (nfx) [inline src/databricks/labs/lsql/dashboards.py:51]: For now, a dashboard can have only one page. But it's going to change in a year

File: src/databricks/labs/lsql/dashboards.py [hunk 2]
```diff
@@ -24,41 +25,59 @@
...
+            with (local_path / f"{page.name}.yml").open("w") as f:
+                yaml.safe_dump(page.as_dict(), f)
+
+    @staticmethod
+    def format_query(query: str) -> str:
```
Comment (nfx) [inline src/databricks/labs/lsql/dashboards.py:56]: Method has to be private, bring back the original name

File: src/databricks/labs/lsql/dashboards.py [hunk 3]
```diff
@@ -24,41 +25,59 @@
...
+        try:
+            parsed_query = sqlglot.parse(query)
+        except sqlglot.ParseError:
+            formatted_query = query
+        else:
```
Comment (nfx) [inline src/databricks/labs/lsql/dashboards.py:61]: This else i'm struggling to follow  else in try-except-else block. Can you move the else body to try?

File: src/databricks/labs/lsql/dashboards.py [hunk 4]
```diff
@@ -24,41 +25,59 @@
...
+        for page in dashboard.pages:
+            with (local_path / f"{page.name}.yml").open("w") as f:
+                yaml.safe_dump(page.as_dict(), f)
+
+    @staticmethod
+    def format_query(query: str) -> str:
+        try:
+            parsed_query = sqlglot.parse(query)
+        except sqlglot.ParseError:
+            formatted_query = query
```
Comment (nfx) [inline src/databricks/labs/lsql/dashboards.py:60]:
```suggestion
            return query
```
Simplify logic

File: src/databricks/labs/lsql/dashboards.py [hunk 5]
```diff
@@ -99,42 +118,44 @@ def deploy_dashboard(
...
-                    )
-                    f.write(f"{pretty};\n")
-            except sqlglot.ParseError:
-                f.write(sql_query)
-
-    def _replace_names(self, node: _DataclassInstance, better_names: dict[str, str]):
-        # walk evely dataclass instance recursively and replace names
-        if isinstance(node, _DataclassInstance):
-            for field in node.__dataclass_fields__.values():
+    def with_better_names(self, dashboard: Dashboard) -> Dashboard:
```
Comment (nfx) [inline src/databricks/labs/lsql/dashboards.py:121]: Keep this method private

File: src/databricks/labs/lsql/dashboards.py [hunk 6]
```diff
@@ -99,42 +118,44 @@ def deploy_dashboard(
...
+        """Replace names with human-readable names."""
+        datasets, better_names = [], {}
+        for dataset in dashboard.datasets:
```
Comment (nfx) [inline src/databricks/labs/lsql/dashboards.py:124]: The idea of replace names is that you contain all manipulation there - that's why we do recursive traversal. Can you rewrite it as "if isinstance(node, Dataset): ..." and move the logic of this method over there.

File: tests/integration/test_dashboards.py [hunk 7]
```diff
@@ -20,68 +19,23 @@ def dashboard_id(ws, make_random):
...
-    dashboard_client = Dashboards(ws)
-    dashboard = dashboard_client.deploy_dashboard(lakeview_dashboard, dashboard_id=dashboard_id)
+    dashboards = Dashboards(ws)
+    dashboard = dashboards.deploy_dashboard(lakeview_dashboard, dashboard_id=dashboard_id)
 
     assert ws.lakeview.get(dashboard.dashboard_id)
+
+
+def test_dashboard_deploys_dashboard_the_same_as_created_dashboard(ws, dashboard_id):
+    queries = Path(__file__).parent / "dashboards" / "dashboard"
```
Comment (nfx) [inline tests/integration/test_dashboards.py:34]:
```suggestion
    queries = Path(__file__).parent / "dashboards" / "first"
```
Otherwise code reads weird

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/lsql/dashboards.py (hunk 2 above)
Comment (JCZuurmond) [reply src/databricks/labs/lsql/dashboards.py:56]: I made it private, kept this name as it is more accurate as it does not operate on a file

File: tests/integration/test_dashboards.py (hunk 7 above)
Comment (JCZuurmond) [reply tests/integration/test_dashboards.py]: renamed to more explicit

**Review by nfx — APPROVED** [review]
Lgtm

**Pull request merged**